Framework/PythonInterface/inc/MantidPythonInterface/kernel/Environment/ReleaseGlobalInterpreterLock.h

```diff
@@ -0,0 +1,52 @@
+#ifndef MANTID_PYTHONINTERFACE_RELEASEGLOBALINTERPRETERLOCK_H_
+#define MANTID_PYTHONINTERFACE_RELEASEGLOBALINTERPRETERLOCK_H_
+
+#include "MantidPythonInterface/kernel/DllConfig.h"
+#include <boost/python/detail/wrap_python.hpp>
+
+namespace Mantid {
+namespace PythonInterface {
+namespace Environment {
+
+/**
+ * Defines a structure for releaseing the Python GIL
+ * using the RAII pattern. This releases the Python GIL
+ * for the duration of the current scope.
+ */
+class PYTHON_KERNEL_DLL ReleaseGlobalInterpreterLock {
```

Comment:
I think the MSVC compiler warnings are due to the fact that the class is implemented entirely in the header. I'd be tempted to move the definitions to a .cpp file like `GlobalInterpreterLock.cpp`